demos/imagenet_classification_demo/main.cpp

```diff
@@ -0,0 +1,357 @@
+// Copyright (C) 2018-2019 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+/**
+* @brief The entry point the Inference Engine sample application
+* @file classification_sample_async/main.cpp
+* @example classification_sample_async/main.cpp
+*/
+#include <fstream>
+#include <vector>
+#include <queue>
+#include <memory>
+#include <string>
+#include <map>
+#include <condition_variable>
+#include <mutex>
+
+#include <inference_engine.hpp>
+
+#include <format_reader_ptr.h>
+
+#include <samples/common.hpp>
+#include <samples/slog.hpp>
+#include <samples/args_helper.hpp>
+#include <samples/classification_results.h>
+#include <samples/ocv_common.hpp>
+#include <samples/common.hpp>
+
+#include <ext_list.hpp>
+#include <vpu/vpu_plugin_config.hpp>
+#include <cldnn/cldnn_config.hpp>
+
+#include <opencv2/core.hpp>
+#include <opencv2/imgcodecs.hpp>
+#include <opencv2/highgui.hpp>
+
+#include <sys/stat.h>
+#include <ext_list.hpp>
+
+#include "classification_sample_async.h"
+#include "captures.hpp"
+#include "grid_mat.hpp"
+#include "my_functor.hpp"
+
+using namespace InferenceEngine;
+
+ConsoleErrorListener error_listener;
+
+bool ParseAndCheckCommandLine(int argc, char *argv[]) {
+    // ---------------------------Parsing and validation of input args--------------------------------------
+    slog::info << "Parsing input parameters" << slog::endl;
+
+    gflags::ParseCommandLineNonHelpFlags(&argc, &argv, true);
+    if (FLAGS_h) {
+        showUsage();
+        showAvailableDevices();
+        return false;
+    }
+    slog::info << "Parsing input parameters" << slog::endl;
+
+    if (FLAGS_i.empty()) {
+        throw std::logic_error("Parameter -i is not set");
+    }
+
+    if (FLAGS_m.empty()) {
+        throw std::logic_error("Parameter -m is not set");
+    }
+
+    return true;
+}
+
+int main(int argc, char *argv[]) {
+    try {
+        if (!ParseAndCheckCommandLine(argc, argv)) {
+            return 0;
+        }
+            
+        std::vector<std::string> imageNames;
+        parseInputFilesArguments(imageNames);
+
+        //~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~read all imgs~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Comment:
If you are going to support such pipeline stage marks, you should follow the other demos style and use `-` instead of `~`.